pkg/octosts/trust_policy_test.go

```diff
@@ -199,10 +241,48 @@ func TestCheckToken(t *testing.T) {
 			SubjectPattern: "(123|456)",
 		},
 		token: &oidc.IDToken{
-			Issuer:  "https://accounts.google.com",
-			Subject: "123999",
+			Issuer:   "https://accounts.google.com",
+			Subject:  "123999",
+			Audience: []string{"octo-sts.dev"},
 		},
 		wantErr: true,
+	}, {
+		name: "matches one of audience pattern",
+		tp: &TrustPolicy{
+			Issuer:          "https://example.com",
+			Subject:         "blah",
+			AudiencePattern: "(octo|nona)-sts\\.dev",
+		},
+		token: &oidc.IDToken{
+			Issuer:   "https://example.com",
+			Subject:  "blah",
+			Audience: []string{"deka-sts.dev", "nona-sts.dev"},
+		},
+		wantErr: false,
+	}, {
+		name: "matches one of audience",
+		tp: &TrustPolicy{
+			Issuer:          "https://example.com",
+			Subject:         "blah",
+			AudiencePattern: "example.com",
+		},
+		token: &oidc.IDToken{
+			Issuer:   "https://example.com",
+			Subject:  "blah",
+			Audience: []string{"octo-sts.dev", "deka-sts.dev", "example.com", "nona-sts.dev"},
+		},
+		wantErr: false,
+	}, {
+		name: "support boolean claims",
+		tp: &TrustPolicy{
+			Issuer:  "https://example.com",
+			Subject: "blah",
+		},
+		token: &oidc.IDToken{
+			Issuer:  "https://example.com",
+			Subject: "blah",
+		},
+		wantErr: false,
 	}}
 
 	// TODO(mattmoor): Figure out how to test custom claims with IDToken.
```

Comment:
There's a reflection trick I found after we initially built this, which should make this possible!

Let me dig it up

Reply:
https://github.com/chainguard-dev/mono/blob/6856b8058b0a43170f5a49336c97f2fbf0943f1e/registry/internal/fake/token.go#L92-L100

Reply:
Neat and gross! Added a bunch of tests to cover the list - lmk what you think